internal/provider/kubernetes/rbac.go

```diff
@@ -5,3 +5,4 @@ package kubernetes
 
 // RBAC for watched resources of Gateway API controllers.
 // +kubebuilder:rbac:groups="",resources=secrets;services;namespaces,verbs=get;list;watch
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch
```

Comment:
shouldnt this be limited to the `envoy-gateway-system` namespace ?

Reply:
As with the other [watched resources](https://github.com/envoyproxy/gateway/blob/main/internal/provider/kubernetes/rbac.go#L6-L7), it's my understanding a controller [watch](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller#Controller.Watch) is across all namespaces and then an eventhandler/predicate are used to filter the sources, e.g. Deployment.

EG has RBAC config split between the Infra Manager and the Kube provider. In this case, the Kube provider controller is the component that requires a Deployment `watch` verb and not the Infra Manager.

@skriss thoughts?

Reply:
When I add the `watch` and `list` verbes to the Infra Manager RBAC, I still see this bug:
```
$ kubectl get role/infra-manager -n envoy-gateway-system -o yaml
...
  resources:
  - deployments
  verbs:
  - create
  - get
  - list
  - watch
  - update
  - delete
```

EG Logs:
```
E0920 18:48:20.575121       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.24.2/tools/cache/reflector.go:167: Failed to watch *v1.Deployment: failed to list *v1.Deployment: deployments.apps is forbidden: User "system:serviceaccount:envoy-gateway-system:envoy-gateway" cannot list resource "deployments" in API group "apps" at the cluster scope
```

@arkodg `... at the cluster scope` is key to the error message.

Reply:
@danehans that sounds right to me, I don't think it's possible to constraint the informer for a single resource type to just one namespace.

Reply:
rbac can be kept within infra-manager when https://github.com/envoyproxy/gateway/issues/371 is worked on